OpenRA.Mods.Common/Traits/Buildings/ProductionAirdrop.cs

```diff
@@ -52,28 +53,31 @@ public override bool Produce(Actor self, ActorInfo producee, string productionTy
 			var map = owner.World.Map;
 			var aircraftInfo = self.World.Map.Rules.Actors[info.ActorType].TraitInfo<AircraftInfo>();
 
-			CPos startPos;
-			CPos endPos;
+			WPos startPos;
+			WPos endPos;
 			WAngle spawnFacing;
 
+			var bounds = map.Bounds;
+			var diagonalVec = map.ProjectedBottomRight - map.ProjectedTopLeft;
+
 			if (info.BaselineSpawn)
 			{
-				var bounds = map.Bounds;
-				var center = new MPos(bounds.Left + bounds.Width / 2, bounds.Top + bounds.Height / 2).ToCPos(map);
-				var spawnVec = owner.HomeLocation - center;
-				startPos = owner.HomeLocation + spawnVec * (Exts.ISqrt((bounds.Height * bounds.Height + bounds.Width * bounds.Width) / (4 * spawnVec.LengthSquared)));
+				var spawnPos = map.CenterOfCell(owner.HomeLocation);
+				var spawnVec = spawnPos - (WPos.Zero + diagonalVec / 2);
```

Comment:
Actually, something similar happens here with `WPos.Zero`. Does `spawnPos - diagonalVec / 2;` not work the same?

Reply:
I believe that would give a `WPos` instead of a `WVec`. These situations are the one annoying edge case for the deliberately restrictive types I defined, but its a good tradeoff for eliminating all the weird and broken things that code used to do before we had them.

Reply:
Are we sure that the `WPos.Zero` shouldn't be `map.ProjectedTopLeft`? The next calls deal with the map edge, which are based on `ProjectedTopLeft`/`ProjectedBottomRight`, not zero.

Reply:
Fixed